core/translate/emitter.rs

```diff
@@ -177,6 +176,22 @@ pub fn emit_program(
         }
     }
 
+    // No rows will be read from source table loops if there is a constant false condition eg. WHERE 0
+    // however an aggregation might still happen,
+    // e.g. SELECT COUNT(*) WHERE 0 returns a row with 0, not an empty result set
+    let skip_loops_label = if plan.contains_constant_false_condition {
```

Comment:
changed this bytecode impl to align with sqlite

```
sqlite> explain select first_name from users where false;
addr  opcode         p1    p2    p3    p4             p5  comment      
----  -------------  ----  ----  ----  -------------  --  -------------
0     Init           0     8     0                    0   Start at 8
1     Goto           0     7     0                    0   
2     OpenRead       0     2     0     2              0   root=2 iDb=0; users
3     Rewind         0     7     0                    0   
4       Column         0     1     1                    0   r[1]= cursor 0 column 1
5       ResultRow      1     1     0                    0   output=r[1]
6     Next           0     4     0                    1   
7     Halt           0     0     0                    0   
8     Transaction    0     0     3     0              1   usesStmtJournal=0
9     Goto           0     1     0                    0 
```

```
limbo> explain select first_name from users where false;
addr  opcode             p1    p2    p3    p4             p5  comment
----  -----------------  ----  ----  ----  -------------  --  -------
0     Init               0     11    0                    0   Start at 11
1     Goto               0     10    0                    0   
2     OpenReadAsync      0     2     0                    0   table=users, root=2
3     OpenReadAwait      0     0     0                    0   
4     RewindAsync        0     0     0                    0   
5     RewindAwait        0     10    0                    0   Rewind table users
6       Column           0     1     1                    0   r[1]=users.first_name
7       ResultRow        1     1     0                    0   output=r[1]
8     NextAsync          0     0     0                    0   
9     NextAwait          0     6     0                    0   
10    Halt               0     0     0                    0   
11    Transaction        0     0     0                    0   
12    Goto               0     1     0                    0
```

```
sqlite> explain select count(1) from products where false;
addr  opcode         p1    p2    p3    p4             p5  comment      
----  -------------  ----  ----  ----  -------------  --  -------------
0     Init           0     12    0                    0   Start at 12
1     Null           0     1     1                    0   r[1..1]=NULL
2     Goto           0     8     0                    0   
3     OpenRead       0     3     0     0              0   root=3 iDb=0; products
4     Rewind         0     8     0                    0   
5       Integer        1     2     0                    0   r[2]=1
6       AggStep        0     2     1     count(1)       1   accum=r[1] step(r[2])
7     Next           0     5     0                    1   
8     AggFinal       1     1     0     count(1)       0   accum=r[1] N=1
9     Copy           1     3     0                    0   r[3]=r[1]
10    ResultRow      3     1     0                    0   output=r[3]
11    Halt           0     0     0                    0   
12    Transaction    0     0     3     0              1   usesStmtJournal=0
13    Goto           0     1     0                    0 
```

```
limbo> explain select count(*) from products where false;
addr  opcode             p1    p2    p3    p4             p5  comment
----  -----------------  ----  ----  ----  -------------  --  -------
0     Init               0     14    0                    0   Start at 14
1     Goto               0     10    0                    0   
2     OpenReadAsync      0     3     0                    0   table=products, root=3
3     OpenReadAwait      0     0     0                    0   
4     RewindAsync        0     0     0                    0   
5     RewindAwait        0     -2    0                    0   Rewind table products
6       Integer          1     2     0                    0   r[2]=1
7       AggStep          0     2     1     count          0   accum=r[1] step(r[2])
8     NextAsync          0     0     0                    0   
9     NextAwait          0     6     0                    0   
10    AggFinal           0     1     0     count          0   accum=r[1]
11    Copy               1     3     0                    0   r[3]=r[1]
12    ResultRow          3     1     0                    0   output=r[3]
13    Halt               0     0     0                    0   
14    Transaction        0     0     0                    0   
15    Goto               0     1     0                    0 
```

^noticed there's an unresolved label in RewindAwait now. Will fix, although it doesn't matter in practice here since RewindAwait will never evaluate

Reply:
> if plan.contains_constant_false_condition

I eventually want to get rid of this and just derive this information instead of having separate metadata for it, and perhaps go back to what sqlite does (flatten all join condition predicates into the where clause), but thats a bigger refactor